mmv1/templates/terraform/custom_delete/compute_rmig_resize_request_delete.go.tmpl

```diff
@@ -0,0 +1,65 @@
+// Get project id
+project, err := tpgresource.GetProject(d, config)
+if err != nil {
+	return fmt.Errorf("Error fetching project for resize request: %s", err)
+}
+
+// Get cancel url
+var cancelUrl string
+cancelUrl, err = tpgresource.ReplaceVars(d, config, "{{"{{"}}ComputeBasePath{{"}}"}}projects/{{"{{"}}project{{"}}"}}/regions/{{"{{"}}region{{"}}"}}/instanceGroupManagers/{{"{{"}}instance_group_manager{{"}}"}}/resizeRequests/{{"{{"}}name{{"}}"}}/cancel")
+
+if err != nil {
+	return err
+}
+
+// Get delete url
+var deleteUrl string
+deleteUrl, err = tpgresource.ReplaceVars(d, config, "{{"{{"}}ComputeBasePath{{"}}"}}projects/{{"{{"}}project{{"}}"}}/regions/{{"{{"}}region{{"}}"}}/instanceGroupManagers/{{"{{"}}instance_group_manager{{"}}"}}/resizeRequests/{{"{{"}}name{{"}}"}}")
+if err != nil {
+	return err
+}
+
+// Either cancel and delete the request based on current state or delete
+if d.Get("state") == "CREATING" || d.Get("state") == "ACCEPTED" {
```

Comment:
would there be any way to force this to happen in a test? It's okay if not, but if there's some trivial way to do it that would be great in terms of getting additional insurance that this path works.

Reply:
I don't think so. Usually the resize request immediately goes to ACCEPTED state (CREATING is the initial state of resize requests) after the request being made, unless there are some failure on QueuedResource side which will force the request to retry several times and then take it to FAILED state or ACCEPTED state. After that either it will goes to SUCCEEDED state immediately if the resources are available in the corresponding zone/region or will be in ACCEPTED state longer until the resources become available. In prod environment, we can't guarantee of the availability of the resources (sometimes it might be available and sometimes it might be not).

Reply:
👍